### ISSUE TITLE
KeyError: 'public_path'

### ISSUE THREAD

**Issue by JonLaliberte** [nimroddolev/chime_tts#95]
## System Information

version | core-2024.2.5
-- | --
installation_type | Home Assistant OS
dev | false
hassio | true
docker | true
user | root
virtualenv | false
python_version | 3.12.1
os_name | Linux
os_version | 6.1.74-haos
arch | x86_64
timezone | America/New_York
config_dir | /config

<details><summary>Home Assistant Community Store</summary>

GitHub API | ok
-- | --
GitHub Content | ok
GitHub Web | ok
GitHub API Calls Remaining | 5000
Installed Version | 1.34.0
Stage | running
Available Repositories | 1408
Downloaded Repositories | 26

</details>

<details><summary>Home Assistant Cloud</summary>

logged_in | true
-- | --
subscription_expiration | January 31, 2025 at 7:00 PM
relayer_connected | true
relayer_region | us-east-1
remote_enabled | true
remote_connected | true
alexa_enabled | false
google_enabled | false
remote_server | us-east-1-0.ui.nabu.casa
certificate_status | ready
instance_id | a034c9ee5fdd455e8fb11e2b33f4502a
can_reach_cert_server | ok
can_reach_cloud_auth | ok
can_reach_cloud | ok

</details>

<details><summary>Home Assistant Supervisor</summary>

host_os | Home Assistant OS 11.5
-- | --
update_channel | stable
supervisor_version | supervisor-2024.02.1
agent_version | 1.6.0
docker_version | 24.0.7
disk_total | 38.7 GB
disk_used | 28.7 GB
healthy | true
supported | true
board | ova
supervisor_api | ok
version_api | ok
installed_addons | Mosquitto broker (6.4.0), Terminal & SSH (9.9.0), MariaDB (2.6.1), Log Viewer (0.17.0), File editor (5.8.0), ESPHome (2024.2.1), chrony (3.0.1), Z-Wave JS UI (3.4.0), rtlamr2mqtt (2.3.4), Frigate Proxy (1.4), Home Assistant Google Drive Backup (0.112.1), Matter Server (5.3.0), InfluxDB (5.0.0)

</details>

<details><summary>Dashboards</summary>

dashboards | 2
-- | --
resources | 8
views | 13
mode | storage

</details>

<details><summary>Recorder</summary>

oldest_recorder_run | January 31, 2024 at 2:36 PM
-- | --
current_recorder_run | February 28, 2024 at 7:07 PM
estimated_db_size | 1427.77 MiB
database_engine | mysql
database_version | 10.6.12

</details>

<details><summary>Victor Smart-Kill</summary>

api_endpoint_reachable | ok
-- | --

</details>

### Checklist

- [X] I have enabled debug logging for my installation.
- [X] I have filled out the issue template to the best of my ability.
- [X] This issue only contains 1 issue (if you have multiple issues, open one issue for each issue).
- [X] This issue is not a duplicate issue of currently [previous issues](https://github.com/nimroddolev/chime_tts/issues?q=is%3Aissue+label%3A%22Bug%22+)..

### Describe the issue

I know over here: https://github.com/nimroddolev/chime_tts/issues/81 after the first release with that feature, things were working fine. Must have been a subsequent release where something broke?

### Reproduction steps

```yaml
service: chime_tts.say
target:
  entity_id: media_player.kitchen_homepod
data:
  chime_path: chord
  delay: 50
  tts_playback_speed: 150
  volume_level: 0.35
  message: Front door
  tts_platform: cloud
  cache: true
  announce: true
  gender: female
  force_announce: true
```

### Debug logs

```text
2024-02-29 07:00:01.728 DEBUG (MainThread) [custom_components.chime_tts] ----- Chime TTS Say Called. Version v0.13.1-beta13 -----
2024-02-29 07:00:01.729 DEBUG (MainThread) [custom_components.chime_tts.queue_manager] Adding service call to queue
2024-02-29 07:00:01.800 DEBUG (MainThread) [custom_components.chime_tts.helpers]  is_say_url = False
2024-02-29 07:00:01.801 DEBUG (MainThread) [custom_components.chime_tts.helpers] ----- General Parameters -----
2024-02-29 07:00:01.801 DEBUG (MainThread) [custom_components.chime_tts.helpers]  * entity_ids = ['media_player.kitchen_homepod']
2024-02-29 07:00:01.801 DEBUG (MainThread) [custom_components.chime_tts.helpers]  * chime_path = chord
2024-02-29 07:00:01.801 DEBUG (MainThread) [custom_components.chime_tts.helpers]  * cache = True
2024-02-29 07:00:01.801 DEBUG (MainThread) [custom_components.chime_tts.helpers]  * offset = 50.0
2024-02-29 07:00:01.801 DEBUG (MainThread) [custom_components.chime_tts.helpers]  * final_delay = 0.0
2024-02-29 07:00:01.802 DEBUG (MainThread) [custom_components.chime_tts.helpers]  * message = Front door
2024-02-29 07:00:01.802 DEBUG (MainThread) [custom_components.chime_tts.helpers]  * tts_platform = cloud
2024-02-29 07:00:01.802 DEBUG (MainThread) [custom_components.chime_tts.helpers]  * tts_playback_speed = 150.0
2024-02-29 07:00:01.802 DEBUG (MainThread) [custom_components.chime_tts.helpers]  * announce = True
2024-02-29 07:00:01.802 DEBUG (MainThread) [custom_components.chime_tts.helpers]  * volume_level = 0.35
2024-02-29 07:00:01.802 DEBUG (MainThread) [custom_components.chime_tts.helpers]  * join_players = False
2024-02-29 07:00:01.802 DEBUG (MainThread) [custom_components.chime_tts.helpers]  * unjoin_players = False
2024-02-29 07:00:01.803 DEBUG (MainThread) [custom_components.chime_tts.helpers] ----- TTS-Specific Params -----
2024-02-29 07:00:01.803 DEBUG (MainThread) [custom_components.chime_tts.helpers]  * gender = female
2024-02-29 07:00:01.804 DEBUG (MainThread) [custom_components.chime_tts]  - Attempting to retrieve previously cached audio...
2024-02-29 07:00:01.823 ERROR (MainThread) [homeassistant.components.automation.front_door_announce_open] Front door - announce open: Error executing script. Unexpected error for call_service at pos 4: 'public_path'
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 485, in _async_step
    await getattr(self, handler)()
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 723, in _async_call_service_step
    response_data = await self._async_run_long_action(
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 685, in _async_run_long_action
    return long_task.result()
           ^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/core.py", line 2279, in async_call
    response_data = await coro
                    ^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/core.py", line 2316, in _execute_service
    return await target(service_call)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/chime_tts/__init__.py", line 112, in async_say
    result = await queue.add_to_queue(async_say_execute, service, is_say_url)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/chime_tts/queue_manager.py", line 33, in async_process_queue
    result = await asyncio.wait_for(
             ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/asyncio/tasks.py", line 520, in wait_for
    return await fut
           ^^^^^^^^^
  File "/config/custom_components/chime_tts/__init__.py", line 134, in async_say_execute
    audio_dict = await async_get_playback_audio_path(params, options)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/chime_tts/__init__.py", line 642, in async_get_playback_audio_path
    audio_dict = await async_get_cached_audio_data(hass, filepath_hash)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/chime_tts/__init__.py", line 1250, in async_get_cached_audio_data
    is_public = helpers.file_exists_in_directory(path, '/www') or helpers.file_exists_in_directory(_data[PUBLIC_PATH_KEY])
                                                                                                   ~~~~~^^^^^^^^^^^^^^^^^
KeyError: 'public_path'
2024-02-29 07:00:01.837 ERROR (MainThread) [homeassistant.components.automation.front_door_announce_open] While executing automation automation.front_door_announce_open
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/automation/__init__.py", line 666, in async_trigger
    return await self.action_script.async_run(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 1600, in async_run
    return await asyncio.shield(run.async_run())
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 435, in async_run
    await self._async_step(log_exceptions=False)
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 487, in _async_step
    self._handle_exception(
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 512, in _handle_exception
    raise exception
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 485, in _async_step
    await getattr(self, handler)()
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 723, in _async_call_service_step
    response_data = await self._async_run_long_action(
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 685, in _async_run_long_action
    return long_task.result()
           ^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/core.py", line 2279, in async_call
    response_data = await coro
                    ^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/core.py", line 2316, in _execute_service
    return await target(service_call)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/chime_tts/__init__.py", line 112, in async_say
    result = await queue.add_to_queue(async_say_execute, service, is_say_url)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/chime_tts/queue_manager.py", line 33, in async_process_queue
    result = await asyncio.wait_for(
             ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/asyncio/tasks.py", line 520, in wait_for
    return await fut
           ^^^^^^^^^
  File "/config/custom_components/chime_tts/__init__.py", line 134, in async_say_execute
    audio_dict = await async_get_playback_audio_path(params, options)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/chime_tts/__init__.py", line 642, in async_get_playback_audio_path
    audio_dict = await async_get_cached_audio_data(hass, filepath_hash)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/chime_tts/__init__.py", line 1250, in async_get_cached_audio_data
    is_public = helpers.file_exists_in_directory(path, '/www') or helpers.file_exists_in_directory(_data[PUBLIC_PATH_KEY])
                                                                                                   ~~~~~^^^^^^^^^^^^^^^^^
KeyError: 'public_path'
2024-02-29 07:00:02.353 WARNING (MainThread) [asyncio] socket.send() raised exception.
2024-02-29 07:00:02.356 ERROR (MainThread) [pyintesishome] <class 'ConnectionResetError'> Exception. ('Connection lost',) / Connection lost
2024-02-29 07:00:30.165 DEBUG (MainThread) [custom_components.chime_tts] ----- Chime TTS Say Called. Version v0.13.1-beta13 -----
2024-02-29 07:00:30.166 DEBUG (MainThread) [custom_components.chime_tts.queue_manager] Adding service call to queue
2024-02-29 07:00:30.168 DEBUG (MainThread) [custom_components.chime_tts.helpers] ``` is_say_url = False
2024-02-29 07:00:30.169 DEBUG (MainThread) [custom_components.chime_tts.helpers] ----- General Parameters -----
2024-02-29 07:00:30.169 DEBUG (MainThread) [custom_components.chime_tts.helpers]  * entity_ids = ['media_player.kitchen_homepod']
2024-02-29 07:00:30.169 DEBUG (MainThread) [custom_components.chime_tts.helpers]  * chime_path = chord
2024-02-29 07:00:30.170 DEBUG (MainThread) [custom_components.chime_tts.helpers]  * cache = True
2024-02-29 07:00:30.170 DEBUG (MainThread) [custom_components.chime_tts.helpers]  * offset = 50.0
2024-02-29 07:00:30.170 DEBUG (MainThread) [custom_components.chime_tts.helpers]  * final_delay = 0.0
2024-02-29 07:00:30.170 DEBUG (MainThread) [custom_components.chime_tts.helpers]  * message = Front door
2024-02-29 07:00:30.170 DEBUG (MainThread) [custom_components.chime_tts.helpers]  * tts_platform = cloud
2024-02-29 07:00:30.170 DEBUG (MainThread) [custom_components.chime_tts.helpers]  * tts_playback_speed = 150.0
2024-02-29 07:00:30.170 DEBUG (MainThread) [custom_components.chime_tts.helpers]  * announce = True
2024-02-29 07:00:30.170 DEBUG (MainThread) [custom_components.chime_tts.helpers]  * volume_level = 0.35
2024-02-29 07:00:30.170 DEBUG (MainThread) [custom_components.chime_tts.helpers]  * join_players = False
2024-02-29 07:00:30.170 DEBUG (MainThread) [custom_components.chime_tts.helpers]  * unjoin_players = False
2024-02-29 07:00:30.170 DEBUG (MainThread) [custom_components.chime_tts.helpers] ----- TTS-Specific Params -----
2024-02-29 07:00:30.170 DEBUG (MainThread) [custom_components.chime_tts.helpers]  * gender = female
2024-02-29 07:00:30.170 DEBUG (MainThread) [custom_components.chime_tts]  - Attempting to retrieve previously cached audio...
2024-02-29 07:00:30.184 ERROR (MainThread) [homeassistant.components.automation.front_door_announce_open] Front door - announce open: Error executing script. Unexpected error for call_service at pos 4: 'public_path'
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 485, in _async_step
    await getattr(self, handler)()
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 723, in _async_call_service_step
    response_data = await self._async_run_long_action(
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 685, in _async_run_long_action
    return long_task.result()
           ^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/core.py", line 2279, in async_call
    response_data = await coro
                    ^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/core.py", line 2316, in _execute_service
    return await target(service_call)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/chime_tts/__init__.py", line 112, in async_say
    result = await queue.add_to_queue(async_say_execute, service, is_say_url)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/chime_tts/queue_manager.py", line 33, in async_process_queue
    result = await asyncio.wait_for(
             ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/asyncio/tasks.py", line 520, in wait_for
    return await fut
           ^^^^^^^^^
  File "/config/custom_components/chime_tts/__init__.py", line 134, in async_say_execute
    audio_dict = await async_get_playback_audio_path(params, options)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/chime_tts/__init__.py", line 642, in async_get_playback_audio_path
    audio_dict = await async_get_cached_audio_data(hass, filepath_hash)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/chime_tts/__init__.py", line 1250, in async_get_cached_audio_data
    is_public = helpers.file_exists_in_directory(path, '/www') or helpers.file_exists_in_directory(_data[PUBLIC_PATH_KEY])
                                                                                                   ~~~~~^^^^^^^^^^^^^^^^^
KeyError: 'public_path'
2024-02-29 07:00:30.189 ERROR (MainThread) [homeassistant.components.automation.front_door_announce_open] While executing automation automation.front_door_announce_open
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/automation/__init__.py", line 666, in async_trigger
    return await self.action_script.async_run(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 1600, in async_run
    return await asyncio.shield(run.async_run())
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 435, in async_run
    await self._async_step(log_exceptions=False)
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 487, in _async_step
    self._handle_exception(
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 512, in _handle_exception
    raise exception
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 485, in _async_step
    await getattr(self, handler)()
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 723, in _async_call_service_step
    response_data = await self._async_run_long_action(
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 685, in _async_run_long_action
    return long_task.result()
           ^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/core.py", line 2279, in async_call
    response_data = await coro
                    ^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/core.py", line 2316, in _execute_service
    return await target(service_call)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/chime_tts/__init__.py", line 112, in async_say
    result = await queue.add_to_queue(async_say_execute, service, is_say_url)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/chime_tts/queue_manager.py", line 33, in async_process_queue
    result = await asyncio.wait_for(
             ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/asyncio/tasks.py", line 520, in wait_for
    return await fut
           ^^^^^^^^^
  File "/config/custom_components/chime_tts/__init__.py", line 134, in async_say_execute
    audio_dict = await async_get_playback_audio_path(params, options)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/chime_tts/__init__.py", line 642, in async_get_playback_audio_path
    audio_dict = await async_get_cached_audio_data(hass, filepath_hash)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/chime_tts/__init__.py", line 1250, in async_get_cached_audio_data
    is_public = helpers.file_exists_in_directory(path, '/www') or helpers.file_exists_in_directory(_data[PUBLIC_PATH_KEY])
                                                                                                   ~~~~~^^^^^^^^^^^^^^^^^
KeyError: 'public_path'
```

**Comment by nimroddolev** [comment]
Hi @JonLaliberte.

I believe this issue has been resolved with v1.0.0.
Please let me know if it still happens for you with the new version, otherwise we can closed this issue.

Thank you.

**Comment by JonLaliberte** [comment]
Thanks @nimroddolev. I'm still running into problems, but I'm not sure if it's on the ChimeTTS side or if it's the wonkiness of the HomePods. I opened a new issue with the warnings/errors I'm seeing now.
